crates/swc_ecma_minifier/src/compress/optimize/unused.rs

```diff
@@ -178,7 +178,6 @@ impl Optimizer<'_> {
                 && v.usage_count == 0
                 && !v.reassigned
                 && v.property_mutation_count == 0
-                && !v.declared_as_catch_param
```

Comment:
I believe we should still check `declared_as_catch_param`. The problem of related test cases arise because `a` in catch param and `a` in catch body shares same syntax context, which should be fixed in resolve pass.

Reply:
Sorry, I do not quite get your point. Do you mean something related to variable hoisting? I run the related tests and whether removing `var a;` or not leads to the same result: `Uncaught ReferenceError: a is not defined`. Could do please tell me a more comprehensible example?

Reply:
```js
function f() {
    try {
        x();
    } catch (a) {
        var a;
    }
}
```
both a resolve to `#3` which is wrong.

Reply:
You mean the `a` in the catch param and the `a` declared in the catch block is actually different? Although I am still not clear why we should preserve `var a` in this case, I find https://tc39.es/ecma262/#sec-variablestatements-in-catch-blocks. Does this spec mean they are actually not different?
> They do not shadow the same-named bindings introduced by the [CatchParameter](https://tc39.es/ecma262/#prod-CatchParameter)

I also try:
```js
!function f() {
    var a = 1;
    try {
        throw "error"
    } catch (a) {
        var a = 2;
    }
    console.log(a) // 1
}()
```

Reply:
IIRC there was a browser that has a bug related to it. Exactly the 

> They do not shadow the same-named bindings introduced by the [CatchParameter](https://tc39.es/ecma262/#prod-CatchParameter)

thing

Reply:
> there was a browser

Highly likely to be Safari

Reply:
> You mean the `a` in the catch param and the `a` declared in the catch block is actually different? Although I am still not clear why we should preserve `var a` in this case, I find https://tc39.es/ecma262/#sec-variablestatements-in-catch-blocks. Does this spec mean they are actually not different?
> 
> > They do not shadow the same-named bindings introduced by the [CatchParameter](https://tc39.es/ecma262/#prod-CatchParameter)
> 
> I also try:
> 
> ```js
> !function f() {
>     var a = 1;
>     try {
>         throw "error"
>     } catch (a) {
>         var a = 2;
>     }
>     console.log(a) // 1
> }()
> ```

Shadow is what we need. But currently they're considered by SWC as same identifier, which may cause problem.

Reply:
Does this PR trigger the bug? If not, I think it can be fixed by another PR

Reply:
No but I believe we should still check `declared_as_catch_param`

Reply:
Ok I'm wrong. They should be the same context.